pkg/apis/options/cookie.go

```diff
@@ -27,7 +28,8 @@ func cookieFlagSet() *pflag.FlagSet {
 	flagSet.StringSlice("cookie-domain", []string{}, "Optional cookie domains to force cookies to (ie: `.yourcompany.com`). The longest domain matching the request's host will be used (or the shortest cookie domain if there is no match).")
 	flagSet.String("cookie-path", "/", "an optional cookie path to force cookies to (ie: /poc/)*")
 	flagSet.Duration("cookie-expire", time.Duration(168)*time.Hour, "expire timeframe for cookie")
-	flagSet.Duration("cookie-refresh", time.Duration(0), "refresh the cookie after this duration; 0 to disable")
+	flagSet.Duration("cookie-refresh", time.Duration(0), "refresh the cookie after this duration; 0 to disable; -1 to auto")
+	flagSet.Uint8("cookie-refresh-grace-pcnt", 100, "in *auto mode only* cookie will be refreshed when specified % of its lifetime has elapsed")
```

Comment:
I like this concept!

Let's make sure to add an options verification that this is set to 1-100.

NIT: Lets just call it `cookie-refresh-percent`, it isn't really a grace period, it is a percentage of lifespan. Simplifies the variable length through the code too, plus `Pcnt` is an odd abbreviate of `Percent`.

I applaud the attempt to shorten the variable name, I've joked in the past some of our method names are full sentences... (*cough* `createSessionStateFromBearerToken` *cough*)

Reply:
Oh, I see the `validation` implementation now, cool! All set on that front.

Reply:
`cookie-refresh-expiry-percent`?

Reply:
Let's keep it short, `cookie-refresh-percent`.

The other option to potentially think about is a unified parameter -- where a plain number is seconds, but a string with ` %` at the end is a percent. 🤔

Reply:
Plain number makes no sense to me, as the expiry is controlled by the IDP and can change dynamically if IDP has dynamic policies. That's why I implemented percentage to begin with.

Reply:
I'm actually really liking the configuration simplicity and user experience by a single configuration option and we just implement the logic whether it is a TTL or % of lifespan based on the presence of a trailing `%` or not.

Reply:
If it's "TTL" it cannot be "percent" though. The name will be confusing.